### ISSUE TITLE
Training not resulting in proper predictions, also loaded weights problem

### ISSUE THREAD

**Issue by Incblob** [Tony607/keras-image-ocr#1]
Hi, i've two issues with the code i hope you might help me with. 

First: Training for the 25 (20+5) epochs does not result in accurate predictions, i ran this both locally and on the google collab notebook, not changing anything, and both times the predictions were really bad. 
basically the output was maybe a few letter from the text repeated up to the max length.
did you change anything from the training to the notebook? or did you train longer? 

Second: (this might be a bit more on my side, but you might have a clue)
Loading the provided weights on the google collab notebook results in accurate predictions, however loading them locally drops the accuracy. Meaning that it gets most words right, but with a letter being wrong, this applies to beam search as well, and the probabilities for the letters are a bit more "diffuse", not as sharp as the ones on the google collab version.
So basically i'm running the same code, loading the same weights, but having a decrease of accuracy locally. Understandable if you can't help me there, just thought you might have run into something like this before.

Locally I'm on a rather fresh install of Ubuntu, all prerequisites installed, using an anaconda virtualenv. 

Thanks for the help,
Nic

ps: i also tried a fresh anaconda env in case something was interfering, with no change

**Comment by Tony607** [comment]
Hi, I just re-trained the model on Colab, it worked as expected. Maybe you can monitor the evaluated results for each training epoch and see if any abnormal.
In my case, an evaluated image is located at `image_ocr/2018:06:05:01:06:18/e21.png`
You can download an image file from Colab like this:
```python
from google.colab import files
files.download('image_ocr/2018:06:05:01:06:18/e21.png')
```

**Comment by Incblob** [comment]
Hey, thanks for the quick reply, i'm currently training again both locally and on Colab, and at least locally it seems to be training properly... i'll write again when it's finished. 
Thanks again :)

**Comment by Incblob** [comment]
Well, as a colleague of mine always says, the problem is usually sitting between the screen and the chair...
I didn't realize that you were redefining the entire model layer for layer for inference, which means that the weights don't transfer as they would if you would just redefine input and output.... 
I thought naming the layers gave the weights persistence, but of course not if you redefine the same layer, that apparently overwrites the trained weights.
So yeah, loading the trained weight file explicitly gets everything working as it should.
Thanks for the help and sorry for the inconvenience :p 

PS: the load_weight function didn't raise an error when it was pointed to a wrong dir :/
